src/react.ts

```diff
@@ -2,18 +2,17 @@ import { useDebugValue } from 'react'
 import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/shim/with-selector'
 import createStore, {
   EqualityChecker,
-  GetState,
-  SetState,
+  Mutate,
   State,
   StateCreator,
   StateSelector,
   StoreApi,
+  StoreMutatorIdentifier,
 } from './vanilla'
 
-export function useStore<T extends State>(api: StoreApi<T>): T
-export function useStore<T extends State, U>(
-  api: StoreApi<T>,
-  selector: StateSelector<T, U>,
+export function useStore<S extends StoreApi<State>, U = ExtractState<S>>(
+  api: S,
+  selector?: StateSelector<ExtractState<S>, U>,
   equalityFn?: EqualityChecker<U>
 ): U
```

Comment:
We should prefer function overloading here too, as we discussed in [#812](https://github.com/pmndrs/zustand/issues/812#issuecomment-1046324462), no?

Reply:
Ah right